test/common/testlib.py

```diff
@@ -1185,6 +1185,9 @@ def copy_cores(self, title, label=None):
             if m.ssh_reachable:
                 directory = "%s-%s-%s.core" % (label or self.label(), m.label, title)
                 dest = os.path.abspath(directory)
+                # overwrite core dumps from previous retries
```

Comment:
I am not sure about this change.
If I understand it correctly, it means that after test finishes, it removes all cores from previous runs and copies some if there are any, right?
So we get cores only from the last run?

Reply:
More precisely, we only get cores from the last *failed* run (`copy_cores()` is only called on failures). That is an approximation of "Keep cores from failed tests, not successes, skips, or retries", and it also avoids the accidental creation of a "coredumps" subdirectory instead of overwriting/adding to the existing dir.